trio/_highlevel_open_tcp_stream.py

```diff
@@ -242,71 +249,62 @@ def format_host_port(host, port):
 
     reorder_for_rfc_6555_section_5_4(targets)
 
-    targets_iter = iter(targets)
-
     # This list records all the connection failures that we ignored.
     oserrors = []
 
-    # It's possible for multiple connection attempts to succeed at the ~same
-    # time; this list records all successful connections.
-    winning_sockets = []
+    # the socket that we're going to complete with, we need to make
+    # sure this is excluded from any cleanup
+    winning_socket = None
 
     # Sleep for the given amount of time, then kick off the next task and
     # start a connection attempt. On failure, expedite the next task; on
```

Comment:
This comment is a bit inaccurate now that the sleeping/task spawning has been moved into the parent – could you update?